src/controllers/eventseats_controller.go

```diff
@@ -0,0 +1,67 @@
+package controllers
+
+import (
+	"slices"
+	"time"
+
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/models"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/repositories"
+	"github.com/google/uuid"
+)
+
+type EventSeatControllerI interface {
+	GetEventSeats(eventId *uuid.UUID, userId *uuid.UUID) (*[][]models.GetSeatsForSeatSelectorDTO, *[]models.GetSeatsForSeatSelectorDTO, *time.Time, *models.KTSError)
+}
+
+type EventSeatController struct {
+	EventSeatRepo repositories.EventSeatRepoI
+}
+
+func (esc *EventSeatController) GetEventSeats(eventId *uuid.UUID, userId *uuid.UUID) (*[][]models.GetSeatsForSeatSelectorDTO, *[]models.GetSeatsForSeatSelectorDTO, *time.Time, *models.KTSError) {
+	seats, kts_err := esc.EventSeatRepo.GetEventSeats(eventId)
+
+	if kts_err != nil {
+		return nil, nil, nil, kts_err
+	}
+
+	seatRows := make(map[int32][]models.GetSeatsForSeatSelectorDTO)
+	currentUserSeats := []models.GetSeatsForSeatSelectorDTO{}
+	var blockedUntil *time.Time
+
+	for _, seat := range *seats {
```

Comment:
why not a db query for this

Reply:
Thought it was more complicated, and as I have to iterate through it anyway and sort it into appropriate places. And didn't get it to work to put it into custom models, but this should work, right?